dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/cache/internal/ZipFileReaper.java

```diff
@@ -669,28 +681,49 @@ public void notify(String methodName, String text) {
             if ( tc.isDebugEnabled() ) {
                 Tr.debug(tc, methodName + " " + text);
             }
+            // System.out.println("ZFR." + methodName + " " + text);
+
             notify();
         }
 
         public void wait(String methodName, String text) throws InterruptedException {
             if ( tc.isDebugEnabled() ) {
                 Tr.debug(tc, methodName + " Waiting for [ " + text + " ]");
             }
+            // System.out.println("ZFR." + methodName + " Waiting for [ " + text + " ]");
             wait(); // throws InterruptedException
             if ( tc.isDebugEnabled() ) {
                 Tr.debug(tc, methodName + " Waited for [ " + text + " ]");
             }
+            // System.out.println("ZFR." + methodName + " Waited for [ " + text + " ]");
         }
 
         public void waitNS(long waitNs, String methodName, String text) throws InterruptedException {
-            long waitMs = waitNs / ZipCachingProperties.ONE_MILLI_SEC_IN_NANO_SEC;
+            long waitMs = waitNs / ZipCachingProperties.NANO_IN_MILLI;
+            int fracWaitNs = (int) (waitNs - (waitMs * ZipCachingProperties.NANO_IN_MILLI));
+
             if ( tc.isDebugEnabled() ) {
-                Tr.debug(tc, methodName + " Waiting [ " + toAbsSec(waitMs) + " ] for [ " + text + " ]"); 
+                Tr.debug(tc, methodName +
+                    " Waiting [ " + Long.toString(waitMs) + " (ms) " + Integer.toString(fracWaitNs) + " (ns) ]" +
+                    " for [ " + text + " ]");
+            }
+            // System.out.println("ZFR." + methodName +
+            //     " Waiting [ " + Long.toString(waitMs) + " (ms) " + Integer.toString(fracWaitNs) + " (ns) ]" +
+            //     " for [ " + text + " ]");
+
+            if ( (waitMs == 0) && (fracWaitNs == 0) ) {
+                throw new IllegalArgumentException(methodName + ": Invalid zero wait request for [ " + text + " ]");
```

Comment:
Should this be changed to:
if ( (waitMs != 0) || (fracWaitNs != 0) ) {
    wait(waitMs, fractWaitNs);
}

Not sure if we should throw an exception or just not wait if 0 is passed.